system/jlib/jdebug.cpp

```diff
@@ -4420,3 +4425,313 @@ jlib_decl bool printLsOf(unsigned pid)
     return true;
 }
 
+//---------------------------------------------------------------------------------------------------------------------
+// Memory monitoring and core dump functionality
+// Notes on useVforkAndGcore:
+// false: uses fork() and abort(). fork() duplicates parent process memory with COW semantics.
+//        abort() causes the kernel to produce a sparse core file efficiently (quick)
+// true: use vfork(), exec and gcore. vfork() runs in same address space as parent process until the exec.
+//       gcore attaches to the process and walks memory and copies it to a core file (slower).
+void generateCoreDump(bool useVforkAndGcore)
+{
+#ifdef __linux__
+    pid_t parentPid = getpid();
+
+    char gCoreFname[64];
+    char gCorePPid[32];
+    if (useVforkAndGcore)
+    {
+        // prepare before fork() to do bare minimum in child process
+
+        // Build a unique relative filename: core.<parentPid>.<timestamp>
+        time_t now = time(NULL);
+        snprintf(gCoreFname, sizeof(gCoreFname), "core.%d.%ld", (int)parentPid, (long)now);
```

Comment:
Is the current directory ok ?  If not, perhaps could add debug dir to path here.